packages/core/src/util/combinators.ts

```diff
@@ -70,6 +72,10 @@ export const createCombinatorRenderInfos = (
         control,
         rootSchema
       ),
-      label: createLabel(subSchema, subSchemaIndex, keyword),
+      label: createLabel(
+        merge({}, resolvedSubSchema, subSchema),
+        subSchemaIndex,
+        keyword
+      ),
```

Comment:
I think the `merge` is a bit too much ;) I would like to suggest to be a bit more explicit here and just remove the unnecessary `createLabel` method.

```suggestion
      label: subSchema.title ?? resolvedSubSchema.title ?? `${keyword}-${subSchemaIndex}`
```